### ISSUE TITLE
String#to_d does not return partially parsed results for trailing decimal

### ISSUE THREAD

**Issue by ibrahima** [ruby/bigdecimal#131]
Hi! We've been using Ruby 2.3 for a while and are trying to upgrade to Ruby 2.5. One issue that I'm noticing is that, perhaps because of the fix to #51 (in #55), the behavior of `String#to_d` is different than it used to be when the string starts with a number but is not a complete number. The behavior is now inconsistent with the behavior of `String#to_f`

I.e.

```ruby
require 'bigdecimal'
require 'bigdecimal/util'

'1.'.to_d
=> 0.0 # Returns 0.0, but used to return 1.0
'1.'.to_f
=> 1.0
```
The documentation still says:

> initial
>    The initial value, as an Integer, a Float, a Rational, a BigDecimal, or a String.
>    If it is a String, spaces are ignored and unrecognized characters terminate the value.

This works okay if I pass in things like "1 pizza" etc, but I guess this specific case of a number with a decimal and nothing after the decimal no longer works, which is kind of unfortunate for my specific use case.

```ruby
'1.'.to_d == '1.'.to_f
# returns true on 2.3, false on 2.5
```

I'm on Ruby 2.5.5 with BigDecimal 1.3.4.

Thanks!

**Comment by ibrahima** [comment]
Ohhh, I didn't read the other issues correctly - this is a dupe of https://github.com/ruby/bigdecimal/issues/63

**Comment by ibrahima** [comment]
Actually, still confused. I installed BigDecimal v1.4.3 via bundler but in my Rails app, I still am getting the wrong value for `"1.".to_d`

```
2.5.5 (rails)[1] » '1.'.to_d
=> 0.0
```

**Comment by mrkn** [comment]
This was resolved by 1.4.4.
